tests/test_inspector.py

```diff
@@ -101,6 +100,12 @@ def setUpClass(cls):
     def tearDownClass(cls):
         rmtree(cls.tempdir)
 
+    def assertListsEquivalent(self, test_list: list, true_list: list):
+        for member in test_list:
+            self.assertIn(member=member, container=true_list)
+        for member in true_list:
+            self.assertIn(member=member, container=test_list)
```

Comment:
Are you trying to test that the lists have the same elements, but may be in a different order? If so, this is probably better: `assert set(test_list) == set(true_list)` or perhaps you want [`assertCountEqual`](https://docs.python.org/3/library/unittest.html?highlight=assertcountequal#unittest.TestCase.assertCountEqual)

Reply:
`self.assertCountEqual` works perfectly, so I've swapped to that. It does print a lot of results in the output though, so I had to set the `maxDiff` flag to `None` just to view it on some simple cases.

I would prefer the `set` equal way, but unfortunately, casting a list as a set requires all the contained objects to be hashable, which our `InspectorMessages` are not. They do, however, have an equality method set by the `dataclass`, so you are able to test if two `InspectorMessages` are exactly equal (_i.e._, they have equal contents).